docker/build-images

```diff
@@ -29,61 +29,78 @@ OPTIONS:
   -d <repo>   Specify the destination repo to receive the images; implies -p;
               defaults to: "$REGISTRY"
   -c          Use cached layers when building containers; defaults to false
+  -e          docker or podman. Used to build and run the container. Defaults to what is available, preferring podman.
   -v          Enable verbose/debug output
 HELP
 }
 
-while getopts ":pd:cv" opt; do
-    case $opt in
-        h  ) usage; exit 0;;
-        p  ) PUSH='true';;
-        d  ) PUSH='true'
-             REGISTRY="${OPTARG}";;
-        c  ) BUILD_ARGS="$(echo $BUILD_ARGS | sed -e s/--no-cache//g)";;
-        v  ) set -x;;
-        \?)
-          echo "Invalid option: -$OPTARG" >&2
-          usage
-          exit 1
-          ;;
-        :)
-          echo "Option -$OPTARG requires an argument." >&2
-          usage
-          exit 1
-          ;;
-    esac
+while getopts ":pd:ce:v" opt; do
+  case $opt in
+  h)
+    usage
+    exit 0
+    ;;
+  p) PUSH='true' ;;
+  d)
+    PUSH='true'
+    REGISTRY="${OPTARG}"
+    ;;
+  c) BUILD_ARGS="$(echo $BUILD_ARGS | sed -e s/--no-cache//g)" ;;
+  e)
+    if [ "${OPTARG}" != "podman" ] && [ "${OPTARG}" != "docker" ]; then
+      echo >&2 "Not a valid e: ${OPTARG}. podman or docker supported."
+      exit 1
+    fi
+    CONTAINER_ENGINE="${OPTARG}"
+    ;;
+  v) set -x ;;
+  \?)
+    echo "Invalid option: -$OPTARG" >&2
+    usage
+    exit 1
+    ;;
+  :)
+    echo "Option -$OPTARG requires an argument." >&2
+    usage
+    exit 1
+    ;;
+  esac
 done
 
+if [ "$CONTAINER_ENGINE" == "" ]; then
+  if type podman >/dev/null 2>&1; then
+    CONTAINER_ENGINE="podman"
+  else
+    CONTAINER_ENGINE="docker"
+  fi
+fi
+
 shift $(($OPTIND - 1))
 
 # Determine image name
 if [ "$1" != "" ]; then
-    IMAGE=$1
+  IMAGE=$1
 fi
 
 get_image_names() {
-  python -c 'import yaml,sys;y=yaml.safe_load(sys.stdin); print(" ".join(y["services"].keys()))' < $SCRIPT_HOME/docker-compose-build.yml
-}
-
-# tags a docker image with a version
-tag_images() {
-  TAG_IMAGES=${IMAGE:-$(get_image_names)}
-  CP_VERSION="$( git describe | cut -d- -f 2 )"
-  for IMG in ${TAG_IMAGES}; do
-    docker tag $REGISTRY/$IMG:latest $REGISTRY/$IMG:$CP_VERSION
-  done
+  python -c 'import yaml,sys;y=yaml.safe_load(sys.stdin); print(" ".join(y["services"].keys()))' <$SCRIPT_HOME/docker-compose-build.yml
 }
 
 echo "Building images..."
 cd $SCRIPT_HOME
-docker-compose -f docker-compose-build.yml build $BUILD_ARGS $IMAGE && tag_images
```

Comment:
If we're not longer using docker-compose-build.yml let's delete that file entirely. the `get_image_names` method above that is also using it can just list them hardcoded (it's just 2, and we rarely if ever change them)

Reply:
Done